src/test/java/seedu/momentum/logic/commands/FindCommandTest.java

```diff
@@ -57,27 +62,114 @@ public void equals() {
     @Test
     public void execute_zeroKeywords_noProjectFound() {
         String expectedMessage = String.format(MESSAGE_PROJECTS_LISTED_OVERVIEW, 0);
-        NameContainsKeywordsPredicate predicate = preparePredicate(" ");
+        NameContainsKeywordsPredicate predicate = prepareNamePredicate(false, " ");
         FindCommand command = new FindCommand(predicate);
         expectedModel.updateFilteredProjectList(predicate);
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
         assertEquals(Collections.emptyList(), model.getFilteredProjectList());
     }
 
     @Test
-    public void execute_multipleKeywords_multipleProjectsFound() {
+    public void anyMatch_multipleNameKeywords_multipleProjectsFound() {
         String expectedMessage = String.format(MESSAGE_PROJECTS_LISTED_OVERVIEW, 3);
-        NameContainsKeywordsPredicate predicate = preparePredicate("Kurz Elle Kunz");
+        NameContainsKeywordsPredicate predicate = prepareNamePredicate(false, "Kurz Elle Kunz");
         FindCommand command = new FindCommand(predicate);
         expectedModel.updateFilteredProjectList(predicate);
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
         assertEquals(Arrays.asList(CARL, ELLE, FIONA), model.getFilteredProjectList());
     }
 
+    @Test
+    public void allMatch_multipleNameKeywords_noProjectsFound() {
+        String expectedMessage = String.format(MESSAGE_PROJECTS_LISTED_OVERVIEW, 0);
+        NameContainsKeywordsPredicate predicate = prepareNamePredicate(true, "Kurz Elle Kunz");
```

Comment:
Maybe you can store "Kurz Elle Kunz" as a String variable since you used it more than once in this class?